drivers/volume/portworx/portworx.go

```diff
@@ -670,6 +673,94 @@ func (d *portworx) ValidateCreateVolume(volumeName string, params map[string]str
 	return nil
 }
 
+func (d *portworx) ValidateCreateSnapshot(volumeName string, params map[string]string) error {
+	var token string
+	token = d.getTokenForVolume(volumeName, params)
+	if val, hasKey := params["refresh-endpoint"]; hasKey {
+		refreshEndpoint, _ := strconv.ParseBool(val)
+		d.refreshEndpoint = refreshEndpoint
+	}
+	volDriver := d.getVolDriver()
+	_, err := volDriver.SnapshotCreate(d.getContextWithToken(context.Background(), token), &api.SdkVolumeSnapshotCreateRequest{VolumeId: volumeName, Name: volumeName+"_snapshot"})
+	if err != nil {
+		fmt.Printf("error when creating local snapshot is %v", err)
+		return err
+	}
+	return nil
+}
+
+func (d *portworx) ValidateCreateSnapshotUsingPxctl(volumeName string) error {
+	nodes := node.GetStorageDriverNodes()
+	_, err := d.nodeDriver.RunCommandWithoutRetry(nodes[0], fmt.Sprintf(pxctlLocalSnapshotCreate, volumeName, volumeName+"-snapshot"), node.ConnectionOpts{
```

Comment:
better to have a function returning the cmd instead of concating the string here.